src/check_release.rs

```diff
@@ -135,56 +119,58 @@ pub(super) fn run_check_release(
     )?;
     config
         .log_verbose(|config| {
-            config.shell_status(
-                "Starting",
-                format_args!(
-                    "{} checks, {} unnecessary",
-                    queries_to_run.len(),
-                    skipped_queries
-                ),
-            )
+            let current_num_threads = rayon::current_num_threads();
+            if current_num_threads == 1 {
+                config.shell_status(
+                    "Starting",
+                    format_args!(
+                        "{} checks, {} unnecessary",
+                        queries_to_run.len(),
+                        skipped_queries
+                    ),
+                )
+            } else {
+                config.shell_status(
+                    "Starting",
+                    format_args!(
+                        "{} checks, {} unnecessary on {current_num_threads} threads",
+                        queries_to_run.len(),
+                        skipped_queries
+                    ),
+                )
+            }
         })
         .expect("print failed");
+
     let queries_start_instant = Instant::now();
+    let mut all_results = queries_to_run
+        .par_iter()
+        .map(|&(query_id, semver_query)| {
+            let start_instant = std::time::Instant::now();
+            // trustfall::execute_query(...) -> dyn Iterator (without Send)
+            // thus the result must be collect()'ed
+            let results = adapter
+                .run_query(&semver_query.query, semver_query.arguments.clone())?
+                .collect_vec();
+            let time_to_decide = start_instant.elapsed();
+            Ok((
+                query_id.as_str(),
+                semver_query.required_update,
+                time_to_decide,
+                results,
+            ))
+        })
+        .collect::<anyhow::Result<Vec<_>>>()?;
 
-    for (query_id, semver_query) in queries_to_run.iter().copied() {
-        let category = match semver_query.required_update {
-            RequiredSemverUpdate::Major => "major",
-            RequiredSemverUpdate::Minor => "minor",
-        };
+    // print PASS/FAIL summaries
+    all_results.retain(|(query_id, required_update, time_to_decide, results)| {
```

Comment:
nit: I'm not a huge fan of this `.retain()`, for two reasons. It means that we rely on `retain()` to do side-effects, which relies on `retain()` running in a particular way (sequential, synchronous, etc.) and that seems kind of unintuitive and convoluted. We're also mutating `all_results` in a way that explicitly makes its contents no longer match the description `all_results`, which might come back to bite us in a variety of ways in the future.

I think I'd rather have an explicit for-loop here, for the sake of readability and intuitiveness. We can _probably_ avoid a `.clone()` of `results` by capturing a slice in `results_with_errors` instead, but if that's wrong and we can't avoid cloning, that's okay too. The "found errors" path is the rarer case, and users will be happy to wait an extra few ms worth of `memcpy` to know that breaking changes were found.